README.md

```diff
@@ -1,42 +1,179 @@
-* ported from some other Python implementation, e.g. PyPy
-* some modules actually aren't implemented yet and are dummy
-
-Note that the main target of micropython-lib is a "Unix" port of the
-aforementioned fork of MicroPython. Actual system requirements vary per
-module. Majority of modules are compatible with the upstream MicroPython,
-though some may require additional functionality/optimizations present in
-the "advanced" fork. Modules not related to I/O may also work without
-problems on bare-metal ports, not just on "Unix" port (e.g. pyboard).
-
-
-Usage
------
-micropython-lib packages are published on PyPI (Python Package Index),
-the standard Python community package repository: https://pypi.org/ .
-On PyPI, you can search for MicroPython related packages and read
-additional package information. By convention, all micropython-lib package
-names are prefixed with "micropython-" (the reverse is not true - some
-package starting with "micropython-" aren't part of micropython-lib and
-were released by 3rd parties).
-
-Browse available packages [via this
-URL](https://pypi.org/search/?q=&o=&c=Programming+Language+%3A%3A+Python+%3A%3A+Implementation+%3A%3A+MicroPython).
-
-To install packages from PyPI for usage on your local system, use the
-`upip` tool, which is MicroPython's native package manager, similar to
-`pip`, which is used to install packages for CPython. `upip` is bundled
-with MicroPython "Unix" port (i.e. if you build "Unix" port, you
-automatically have `upip` tool). Following examples assume that
+# micropython-lib
+
+micropython-lib is a project to develop a non-monolithic standard library
+for MicroPython. Each module comes from one of the following sources (and thus
+each module has its own licensing terms):
+
+* written from scratch specifically for MicroPython.
+* ported from CPython.
+* ported from some other Python implementation, e.g. PyPy.
+* some modules actually aren't yet implemented and are dummy.
+
+The version of the library in this repository is compatible with
+[official MicroPython](https://github.com/micropython/micropython.git) 
+including firmware builds downloaded from [micropython.org](https://micropython.org/).
+
+Paul Sokolovsky's [Pycopy](https://github.com/pfalcon/micropython) MicroPython
+fork requires [this library version](https://github.com/pfalcon/micropython-lib).
+Some modules in this library are incompatible with official MicroPython. Pycopy
+users should follow installation instructions on the above library site.
+
+The following notes apply to users of official MicroPython.
+
+To aid installation, library modules are available on [PyPi](https://pypi.org/)
+however the normal `pip` and `pip3` tools cannot be used. MicroPython provides
+a `upip.py` version optimised to run on microcontrollers. The library modules
+on PyPi are those compatible with the Pycopy fork. Where those are incompatible
+with official MicroPython, the official version of `upip` will ensure that the
```

Comment:
I am the author of upip, and "official" version comes from me. So, please stop calling vendor forks "official", call that particular fork what it is - "George Robotics Ltd. fork".